### PR TITLE
Update scrolling behaviour

### PR DESCRIPTION
**Pull request by 30mar**
## Description
update scrolling behavior for pandemic tile and adding new local statistic region

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-14116

## Screenshots
https://user-images.githubusercontent.com/15270737/195104858-2db0eb26-2c51-4326-a635-6130b97e3c82.mov

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsCellModel.swift [hunk 1]
```diff
@@ -48,10 +50,12 @@ class HomeStatisticsCellModel {
 	// MARK: - Internal
 	
 	/// The default set of 'global' statistics for every user
+	private(set) var hasFreshRegion = false
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsCellModel.swift:53]:
```suggestion
	private(set) var hasNewRegion = false
```

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsCellModel.swift [hunk 2]
```diff
@@ -31,9 +31,11 @@ class HomeStatisticsCellModel {
 			.store(in: &subscriptions)
 		
 		localStatisticsProvider.regionStatisticsData
-			.sink { [weak self] regionStatisticsData in
-				  self?.regionStatisticsData = regionStatisticsData
-				  Log.debug("Updating local statistics cell model. \(private: "\(self?.regionStatisticsData.count ?? -1)")", log: .localStatistics)
+			.sink { [weak self] newRegionStatisticsData in
+				guard let self = self else { return }
+				self.hasFreshRegion = newRegionStatisticsData.count > self.regionStatisticsData.count
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsCellModel.swift:36]:
```suggestion
				self.hasNewRegion = newRegionStatisticsData.count > self.regionStatisticsData.count
```

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift [hunk 3]
```diff
@@ -31,13 +31,17 @@ class HomeStatisticsTableViewCell: UITableViewCell {
...
+			scrollView.scrollRectToVisible(stackView.arrangedSubviews[safe: 1]?.frame ?? .zero, animated: true)
+		}
 		// Scroll to first statistics card initially if local statistics exist, and when entering/leaving edit mode
 		if let cellModel = cellModel, !cellModel.regionStatisticsData.isEmpty,
 			scrollView.bounds.origin.x == 0,
 			let firstStatisticsCard = stackView.arrangedSubviews[safe: 1],
 			let secondStatisticsCard = stackView.arrangedSubviews[safe: 2] {
 			DispatchQueue.main.async { [weak self] in
-				self?.scrollView.scrollRectToVisible(HomeStatisticsTableViewCell.editingStatistics ? secondStatisticsCard.frame : firstStatisticsCard.frame, animated: self?.wasAlreadyShown ?? false)
+				let shouldScrollToLocalStatistics = HomeStatisticsTableViewCell.editingStatistics || cellModel.hasFreshRegion
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift:43]:
```suggestion
				let shouldScrollToLocalStatistics = HomeStatisticsTableViewCell.editingStatistics || cellModel.hasNewRegion
```

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/Model/LocalStatisticsRegion.swift [hunk 4]
```diff
@@ -9,7 +9,7 @@ struct LocalStatisticsRegion: Codable, Equatable {
 	let name: String
 	let id: String
 	let regionType: RegionType
-	
+	var justAdded: Bool = false
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/Model/LocalStatisticsRegion.swift:12]: Do we need this?

**Review by naveeddotio — APPROVED** [review]

**Review by flxschmidt — APPROVED** [review]

**Review by puneetmahali1 — APPROVED** [review]

**Pull request merged**